gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java

```diff
@@ -0,0 +1,402 @@
+        .put("binary_float", "float").put("binary_double", "double").put("date", "date").put("datetime", "timestamp")
+        .put("timestamp", "timestamp").put("timestamp with time zone", "timestamp").put("numeric", "double")
+        .put("timezone with local timezone", "timestamp").put("raw", "string").put("long raw", "string")
+        .put("rowid", "string").put("urowid", "string").put("xmltype", "string").put("float", "float")
+        .put("dec", "double").put("decimal", "double").put("integer", "int").put("int", "int")
+        .put("smallint", "int").put("real", "double").put("double precision", "double")
+        .put("interval year", "date").put("interval day", "timestamp").put("timestamp(0)", "timestamp")
+        .put("timestamp(1)", "timestamp").put("timestamp(2)", "timestamp").put("timestamp(3)", "timestamp")
+        .put("timestamp(4)", "timestamp").put("timestamp(5)", "timestamp").put("timestamp(6)", "timestamp")
+        .put("timestamp(7)", "timestamp").put("timestamp(8)", "timestamp").put("timestamp(9)", "timestamp").build();
+    return dataTypeMap;
+  }
+
+  @Override
+  public Iterator<JsonElement> getRecordSetFromSourceApi(String schema, String entity, WorkUnit workUnit,
+      List<Predicate> predicateList) throws IOException {
+    return null;
+  }
+
+  @Override
+  public String getConnectionUrl() {
+    String host = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_HOST_NAME);
+    String port = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_PORT);
+    String sid = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_SID).trim();
+    String url = "jdbc:oracle:thin:@" + host.trim() + ":" + port + ":" + /*host.trim().split("\\.")[0]*/ sid;
+    return url;
+  }
+
+  @Override
+  public long exractSampleRecordCountFromQuery(String query) {
+    if (StringUtils.isBlank(query)) {
+      return SAMPLERECORDCOUNT;
+    }
+
+    long recordcount = SAMPLERECORDCOUNT;
+    String inputQuery = query.toLowerCase();
+
+    int limitStartIndex = -1;
+
+    boolean multiPredicate = inputQuery.indexOf(" and ") != -1 ? true : false;
+    boolean leadingLimit = false;
+    if (!multiPredicate) {
+      limitStartIndex = inputQuery.indexOf(" where rownum <= ");
+    } else {
+      limitStartIndex = inputQuery.indexOf(" and rownum <= ");
+      if (limitStartIndex == -1 && inputQuery.indexOf(" where rownum <= ") > -1) {
+        limitStartIndex = inputQuery.indexOf(" where rownum <= ");
+        leadingLimit = true;
+      } 
+    }
+
+    int limitEndIndex = getLimitEndIndex(inputQuery, limitStartIndex);
+    if (limitStartIndex > 0) {
+      String limitValue = query.substring(query.indexOf("<=") + 3, limitEndIndex);
+      try {
+        recordcount = Long.parseLong(limitValue);
+      } catch (Exception e) {
+        log.error("Ignoring incorrct limit value in input query:" + limitValue);
+      }
+    }
+    return recordcount;
+  }
```

Comment:
I know that queries are currently built / modified by string manipulation but I think it would be less error prone to match the input query against a regex that contains the _where rownum_.. clause. Getting the recordCount would be easier as well as removing the sample clause.